src/posix.c

```diff
@@ -216,9 +217,48 @@ int p_write(git_file fd, const void *buf, size_t cnt)
 	return 0;
 }
 
-#ifdef NO_MMAP
+ssize_t p_pread(git_file fd, void *buf, size_t nbyte, git_off_t offset)
+{
+#ifdef GIT_WIN32
+	/* TODO: Implement using Win32 overlapped IO to avoid updating fd */
 
-#include "map.h"
+	ssize_t error;
+	const git_off_t cpos = p_lseek(fd, 0, SEEK_CUR);
+
+	if ((error = p_lseek(fd, offset, SEEK_SET) < 0 ? -1 : 0))
```

Comment:
I had to read this a few times - I think that we could simplify its understandability, at the expense of an extra line or two.  

```
if (p_lseek(fd, offset, SEEK_SET) < 0) {
    error = -1;
    goto cleanup;
}
```

But since we don't really want to propagate the error value, we want to return `-1` on failure, then it might make sense to have an `on_error` label that just returns `-1`.